src/main/src/main/java/org/geoserver/security/decorators/DefaultSecureDataFactory.java

```diff
@@ -114,14 +115,16 @@ public Object secure(Object object, WrapperPolicy policy) {
         } else if (FeatureIterator.class.isAssignableFrom(clazz)) {
             return new SecuredFeatureIterator((FeatureIterator) object); 
         }
-        
+
         // try coverage readers and formats
         if(GridCoverage2DReader.class.isAssignableFrom(clazz)) {
             return new SecuredGridCoverage2DReader((GridCoverage2DReader) object, policy);
+        } else if(StructuredGridCoverage2DReader.class.isAssignableFrom(clazz)) {
```

Comment:
Right, this remembers me why I put the structured in first place 2 weeks ago when doing the local patch. Fixing it.